primer3/thermoanalysis.pyx

```diff
@@ -219,41 +224,118 @@ cdef class ThermoResult:
         }
 
 
+def _try_get_dict_int(
+        argname: str,
+        key: Union[str, int],
+        dict_obj: Dict[str, int],
+) -> int:
+    '''Helper function to translate an string name to an integer arguments
+    using a dictionary:
```

Comment:
This was a bit of a struggle for me, could something like `Resolve enum value using either the string or int form` work?

Reply:
(I realize it's meant to be more generic that that so maybe a simple rewording would suffice)

Reply:
Your suggestion works for me

Reply:
updated variable names and changed function name to `_conditional_get_enum_int`